snapcraft/internal/deltas/_deltas.py

```diff
@@ -84,6 +87,15 @@ def _check_delta_gen_tool(self):
         if not file_utils.executable_exists(self.delta_tool_path):
             raise DeltaToolError(delta_tool=self.delta_tool_path)
 
+    def _check_delta_size_constraint(self, delta_path):
+        """Ensure delta is sufficiently smaller than target snap."""
```

Comment:
90% is a bit arbitrary, but it is the threshold that was established for delta downloads. The idea here being that although there is still some bandwidth saved, there is still a cost for delta application in the delta service, and pushing the full snap is _probably_ faster in that case. We could certainly spend more time benchmarking to have more concrete data to support this.